CalibTracker/SiPixelESProducers/plugins/alpaka/SiPixelCablingSoAESProducer.cc

```diff
@@ -0,0 +1,157 @@
+#include "CondFormats/SiPixelObjects/interface/SiPixelFedCablingTree.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelQuality.h"
+#include "DataFormats/SiPixelClusterSoA/interface/ClusteringConstants.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelMappingHost.h"
+#include "CondFormats/SiPixelObjects/interface/alpaka/SiPixelMappingDevice.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  class SiPixelCablingSoAESProducer : public ESProducer {
+  public:
+    SiPixelCablingSoAESProducer(edm::ParameterSet const& iConfig)
+        : ESProducer(iConfig), useQuality_(iConfig.getParameter<bool>("UseQualityInfo")) {
+      auto const& component = iConfig.getParameter<std::string>("ComponentName");
+      auto cc = setWhatProduced(this, component);
+      cablingMapToken_ = cc.consumes(edm::ESInputTag{"", iConfig.getParameter<std::string>("CablingMapLabel")});
+      if (useQuality_) {
+        qualityToken_ = cc.consumes();
+      }
+      geometryToken_ = cc.consumes();
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<std::string>("ComponentName", "");
+      desc.add<std::string>("CablingMapLabel", "")->setComment("CablingMap label");
+      desc.add<bool>("UseQualityInfo", false);
+      descriptions.addWithDefaultLabel(desc);
+    }
+
+    std::unique_ptr<SiPixelMappingHost> produce(const CkfComponentsRecord& iRecord) {
+      auto cablingMap = iRecord.getTransientHandle(cablingMapToken_);
+
+      const SiPixelQuality* quality = nullptr;
+      if (useQuality_) {
+        auto qualityInfo = iRecord.getTransientHandle(qualityToken_);
+        quality = qualityInfo.product();
+      }
+
+      bool hasQuality = quality != nullptr;
+      auto geom = iRecord.getTransientHandle(geometryToken_);
+      auto product = std::make_unique<SiPixelMappingHost>(
+          pixelgpudetails::MAX_SIZE,
+          cms::alpakatools::
+              host());  //(pixelgpudetails::MAX_SIZE, *(cablingMap.product()), hasQuality, cms::alpakatools::host());
```

Comment:
Is the comment still necessary? If yes, consider moving to preceding line for better formatting.